path-aware/provider.go

```diff
@@ -0,0 +1,120 @@
+/*
```

Comment:
I'm wondering if this whole "path-aware provider" shouldn't be moved in with the example? Are users  supposed to use this as-is?

Reply:
It almost full blown provider. I dont see why users should be able to use it "as-is".